core/Plugin/Visualization.php

```diff
@@ -461,10 +461,30 @@ private function postDataTableLoadedFromAPI()
         if ($this->dataTable instanceof DataTable) {
             $this->metadata = $this->dataTable->getAllTableMetadata();
 
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $metaData = $item->getAllTableMetadata();
+                    // if ts_created not exist exit loop
+                    if (!isset($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        break;
+                    }
+                    // if ts_created not record in the metadata or if current is more recent up the metadata
+                    if (!isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])
+                      || strtotime($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        $this->metadata = $metaData;
+                    }
+               }
             }
         }
+        if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+            $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
```

Comment:
I would suggest to pass `$metadata[DataTable::ARCHIVED_DATE_METADATA_NAME]` as parameter to `makePrettyArchivedOnText` instead of accessing the private property in that method. As the metadata property only seems to be used for that one special case, it can be removed afterwards.